instrumentation/opentelemetry-instrumentation-pymssql/src/opentelemetry/instrumentation/pymssql/__init__.py

```diff
@@ -0,0 +1,180 @@
+
+        user = kwargs.get("user") or connect_method_args.user
+        if user is not None:
+            self.span_attributes["db.user"] = user
+
+        port = kwargs.get("port") or connect_method_args.port
+        host = kwargs.get("server") or connect_method_args.server
+        if host is None:
+            host = kwargs.get("host") or connect_method_args.host
+            if host is not None:
+                # The host string can include the port, separated by either a coma or
+                # a column
+                for sep in (":", ","):
+                    if sep in host:
+                        tokens = host.rsplit(sep)
+                        host = tokens[0]
+                        if len(tokens) > 1:
+                            port = tokens[1]
+        if host is not None:
+            self.span_attributes["net.peer.name"] = host
+        if port is not None:
+            self.span_attributes["net.peer.port"] = port
+
+        charset = kwargs.get("charset") or connect_method_args.charset
+        if charset is not None:
+            self.span_attributes["db.charset"] = charset
+
+        tds_version = (
+            kwargs.get("tds_version") or connect_method_args.tds_version
+        )
+        if tds_version is not None:
+            self.span_attributes["db.protocol.tds.version"] = tds_version
+
+        return dbapi.get_traced_connection_proxy(connection, self)
+
+
+
+class PyMSSQLInstrumentor(BaseInstrumentor):
+
+    def instrumentation_dependencies(self) -> Collection[str]:
+        return _instruments
+
+    def _instrument(self, **kwargs):
+        """Integrate with the pymssql library.
+        https://github.com/pymssql/pymssql/
+        """
+        tracer_provider = kwargs.get("tracer_provider")
+
+        dbapi.wrap_connect(
+            __name__,
+            pymssql,
+            "connect",
+            _DATABASE_SYSTEM,
+            version=__version__,
+            tracer_provider=tracer_provider,
+            # pymssql does not keep the connection attributes in its connection object;
+            # instead, we get the attributes from the connect method (which is done
+            # via PyMSSQLDatabaseApiIntegration.wrapped_connection)
+            db_api_integration_factory=PyMSSQLDatabaseApiIntegration,
+        )
+
+    def _uninstrument(self, **kwargs):
+        """ "Disable pymssql instrumentation"""
+        dbapi.unwrap_connect(pymssql, "connect")
+
+    @staticmethod
+    def instrument_connection(connection, tracer_provider=None):
+        """Enable instrumentation in a pymssql connection.
+
+        Args:
+            connection: The connection to instrument.
+            tracer_provider: The optional tracer provider to use. If omitted
+                the current globally configured one is used.
+
+        Returns:
+            An instrumented connection.
+        """
+
+        return dbapi.instrument_connection(
+            __name__,
```

Comment:
How come we use `DatabaseApiIntegration` when instrumenting a connection and not `PyMSSQLDatabaseApiIntegration`?

Reply:
I will look into adding a db_api_integration_factory paramter to dbapi.instrument_connection.